### DISCUSSION TITLE
Motion Blur:  Detecting, classifying & tracking vehicles in videos suffering Motion Blur

### DISCUSSION THREAD

**Discussion by zburq** [ultralytics/ultralytics#12846] [Q&A]
Al all,

I have trained YOLO V8 models on some manually labelled videos of traffic in order to detect, classify & track the vehicles.  The model has been generally successful on good quality video. 

Now I am faced with a video in poor lighting conditions, where the shutter speed automatically slowed down, and the vehicles are blurred.  The camera is looking at the traffic side-on (perpendicularly) which maximizes the relative orthogonal speed and motion blur. 

Are you aware of any techniques to pre-process these videos to reduce/eliminate the motion blur?

Any other ideas to deal with situations like this?

Many thanks,

ZBurq

**Comment by pderrenger** [comment]
Hi ZBurq,

Dealing with motion blur, especially in low-light conditions, can indeed be challenging. Here are a couple of suggestions that might help improve the detection performance:

1. **Pre-processing with Deblurring**: You can use OpenCV or other libraries to apply deblurring techniques such as Wiener filter or Blind Deconvolution before feeding the frames to your YOLOv8 model. Here's a basic example using OpenCV:

   ```python
   import cv2

   def deblur_image(image_path):
       img = cv2.imread(image_path)
       deblur = cv2.GaussianBlur(img, (5, 5), 0)
       return deblur

   # Example usage
   processed_frame = deblur_image('path_to_blurry_frame.jpg')
   ```

2. **Adjusting Model Confidence**: Sometimes, simply lowering the confidence threshold can help in detecting objects in blurred images, though it might increase false positives.

3. **Training on Blurred Images**: If possible, include some pre-blurred images in your training dataset. This can help the model learn to recognize features in blurred conditions.

4. **Frame Averaging**: If the video has a high frame rate, averaging multiple frames can sometimes help reduce the effect of motion blur.

I hope these suggestions help you enhance vehicle detection in challenging conditions! Let me know if you have any more questions.
